.github/workflows/ci.yaml

```
@@ -0,0 +1,31 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  tests:
+    name: "Tests on Python ${{ matrix.python-version }} for ${{ matrix.runs-on }}"
+    runs-on: "${{ matrix.runs-on }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.12"]
+        runs-on: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        include:
+          - python-version: "3.11"
+            runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ matrix.python-version }}"
+
+      - name: Install package and test deps
+        run: pip install -e .[test]
```

Change the installation of the pytest
```suggestion
        run: pip install pytest
```